rtl/eclrtl/eclrtl.cpp

```diff
@@ -3520,8 +3538,26 @@ hash64_t rtlHash64VUnicode(UChar const * k, hash64_t initval)
 unsigned rtlHash32Data(size32_t len, const void *buf, unsigned hval)
 {
     const unsigned char *bp = (const unsigned char *)buf;   /* start of buffer */
-    const unsigned char *be = bp + len;     /* beyond end of buffer */
 
+#if __BYTE_ORDER == __LITTLE_ENDIAN
+    //This possibly breaks the aliasing rules for c++, but I can't see it causing any problems
+    while (len >= sizeof(unsigned))
```

Comment:
You're right it does - I should implement as a loop, and let it unroll.
I might time the difference for unsigned and unsigned8.  The bottleneck is the multiplies (on x86) which take 10 cycles each, so I suspect it won't speed it up by that much.